### PR TITLE
Start docs work for FIM

### PR DESCRIPTION
**Pull request by arirubinstein**
Initial work on the FIM documentation, including example configs, and tuning help for linux.  Addresses issue #1014

### CODE REVIEW

**Comment by osqueryer** [comment]
Can one of the osquery admins reply with "ok to test" to kick off a build...

**Comment by theopolis** [comment]
Looks great!

FYI: As soon as we cut the next version I'll switch RTD to use GH versioning, instead of building everything as "latest".

File: docs/wiki/deployment/file-integrity-monitoring.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    {
+      "name": "crontab",
+      "query": "select * from crontab;",
+      "interval": 300
+    },
+    {
+      "name": "file_changes",
+      "query": "select * from file_changes;",
+      "interval": 300
+    }
```
Comment (marpaia) [inline docs/wiki/deployment/file-integrity-monitoring.md]: we should update all new docs to use the "new" config schedule syntax: https://osquery.readthedocs.org/en/latest/introduction/using-osqueryd/#configuration-and-query-schedule

File: docs/wiki/deployment/file-integrity-monitoring.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  ],
+  "additional_monitoring": {
```
Comment (marpaia) [inline docs/wiki/deployment/file-integrity-monitoring.md]: we also changed this from being a key in `additional_monitoring` to being a top-level key of it's own, but for the life of me, I can't remember what it is.

Comment (marpaia) [inline docs/wiki/deployment/file-integrity-monitoring.md]: found it https://github.com/facebook/osquery/blob/master/osquery/config/config.cpp#L209

Comment (theopolis) [inline docs/wiki/deployment/file-integrity-monitoring.md]: The `"additional_monitoring"` key was verbose+redundant, if it's included in a config JSON all it's subkeys are appended to the root: https://github.com/facebook/osquery/blob/master/osquery/config/config.cpp#L184

File: docs/wiki/deployment/file-integrity-monitoring.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+As file changes happen, events will appear in the `file_changes` table.  During a file_change event, the md5, sha1, and sha256 for the file will be calculated if possible.  A sample event looks like this:
+
+```json
+{
+  "action":"ATTRIBUTES_MODIFIED",
+  "category":"homes",
+  "md5":"bf3c734e1e161d739d5bf436572c32bf",
+  "sha1":"9773cf934440b7f121344c253a25ae6eac3e3182",
+  "sha256":"d0d3bf53d6ae228122136f11414baabcdc3d52a7db9736dd256ad81229c8bfac",
+  "target_path":"\\\/root\\\/.ssh\\\/authorized_keys",
```
Comment (marpaia) [inline docs/wiki/deployment/file-integrity-monitoring.md]: We fixed this bug with double encoding a few weeks ago I believe. If you generate some logs using a binary built from master, this line would be `\/root\/.ssh\/authorized_key`

**Comment by arirubinstein** [comment]
Cool, thanks for the comments.  I'll rework this tomorrow with the updated schema and notes, and re-pull.

**Comment by marpaia** [comment]
Feel free to leave the PR up and iterate on the PR itself by adding more commits to your branch. When you commit to `arirubinstein:master`, this PR will automagically get updated.

If started a new branch/PR is more aligned with your workflow though, feel free to do that too, whatever is easiest for you.

**Pull request closed without merging**